### PR TITLE
ES6: Convert bootstraps/enhanced/trail

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

ES6: Converts `bootstraps/enhanced/trail`. I also added a bit of fastdom around the expensive reads.

### CODE REVIEW

**Comment by SiAdcock** [comment]
Nice one. Would be good to type-annotate the function sigs?

File: static/src/javascripts/bootstraps/enhanced/trail.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        const el = qwery(selector)[0];
+
+        if (el) {
+            proximityLoader.add(el, 1500, insert);
+        }
+    }
+};
+
+const initPopular = () => {
+    if (!config.get('page.isFront')) {
```
Comment (SiAdcock) [inline static/src/javascripts/bootstraps/enhanced/trail.js]: OMG thanks for the `get`s! 🙏

File: static/src/javascripts/bootstraps/enhanced/trail.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+const initOnwardContent = () => {
+    insertOrProximity('.js-onward', () => {
+        if (
+            (config.get('page.seriesId') || config.get('page.blogIds')) &&
+            config.get('page.showRelatedContent')
+        ) {
+            new OnwardContent(qwery('.js-onward'));
+        } else if (config.get('page.tones') !== '') {
```
Comment (SiAdcock) [inline static/src/javascripts/bootstraps/enhanced/trail.js]: Worth passing a [`defaultValue`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/lib/config.js#L10) of empty string to catch possibility of `config.page.tones` being `undefined`?

File: static/src/javascripts/bootstraps/enhanced/trail.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                    [...els].forEach(c => {
+                        new TonalComponent().fetch(c, 'html');
+                    });
+                });
+        }
+    });
+};
+
+const initDiscussion = () => {
+    if (config.switches.enableDiscussionSwitch && config.page.commentable) {
```
Comment (SiAdcock) [inline static/src/javascripts/bootstraps/enhanced/trail.js]: Maybe we can use `config.get()` here too!

File: static/src/javascripts/bootstraps/enhanced/trail.js (hunk 2 above)
Comment (gustavpursche) [reply static/src/javascripts/bootstraps/enhanced/trail.js]: Seems like a sensible change to me.

**Review by SiAdcock — APPROVED** [review]
Great!

**Pull request merged**